nimble/host/services/audio/pacs/src/ble_audio_svc_pacs.c

```diff
@@ -0,0 +1,428 @@
+};
+
+static int
+codec_record_to_pacs_entry(struct ble_audio_codec_record *record, void *arg)
+{
+    struct pac_read_cb_arg *cb_arg = (struct pac_read_cb_arg *)arg;
+    int rc;
+
+    if (!(record->flags & cb_arg->flags)) {
+        return 0;
+    }
+
+    rc = os_mbuf_append(cb_arg->om, &record->codec_id,
+                        sizeof(struct ble_audio_codec_id));
+    if (rc) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    rc = os_mbuf_append(cb_arg->om, &record->codec_spec_caps_len,
+                   sizeof(uint8_t));
+    if (rc) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    rc = os_mbuf_append(cb_arg->om, &record->codec_spec_caps, record->codec_spec_caps_len);
+    if (rc) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    rc = os_mbuf_append(cb_arg->om, &record->metadata_len,
+                   sizeof(uint8_t));
+    if (rc) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    rc = os_mbuf_append(cb_arg->om, &record->metadata, record->metadata_len);
+    if (rc) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    cb_arg->pac_count++;
+
+    return 0;
+}
+
+static int
+ble_svc_audio_pacs_sink_pac_read_access(struct ble_gatt_access_ctxt *ctxt)
+{
+    struct pac_read_cb_arg cb_arg = {
+        .flags = BLE_AUDIO_CODEC_FLAG_SINK,
+        .om = ctxt->om,
+        .pac_count = 0
+    };
+    int rc;
+    uint8_t *pac_count;
+
+    pac_count = os_mbuf_extend(ctxt->om, sizeof(uint8_t));
+    rc = ble_audio_codec_foreach(codec_record_to_pacs_entry, (void *)&cb_arg);
+
+    *pac_count = cb_arg.pac_count;
+
+    return rc == 0 ? 0 : BLE_ATT_ERR_INSUFFICIENT_RES;
+}
+
+static int
+ble_svc_audio_pacs_source_pac_read_access(struct ble_gatt_access_ctxt *ctxt)
+{
+    struct pac_read_cb_arg cb_arg = {
+        .flags = BLE_AUDIO_CODEC_FLAG_SOURCE,
+        .om = ctxt->om,
+        .pac_count = 0
+    };
+    int rc;
+    uint8_t *pac_count;
+
+    pac_count = os_mbuf_extend(ctxt->om, sizeof(uint8_t));
+    rc = ble_audio_codec_foreach(codec_record_to_pacs_entry, (void *)&cb_arg);
+
+    *pac_count = cb_arg.pac_count;
+
+    return rc == 0 ? 0 : BLE_ATT_ERR_INSUFFICIENT_RES;
+}
+
+static int
+ble_svc_audio_pacs_sink_audio_loc_read_access(struct ble_gatt_access_ctxt *
+                                              ctxt)
+{
+    int rc;
+
+    rc = os_mbuf_append(ctxt->om, &ble_svc_audio_pacs_sink_audio_locations,
+                        sizeof ble_svc_audio_pacs_sink_audio_locations);
+
+    return rc == 0 ? 0 : BLE_ATT_ERR_INSUFFICIENT_RES;
+}
+
+static int
+ble_svc_audio_pacs_source_audio_loc_read_access(struct ble_gatt_access_ctxt *
+                                                ctxt)
+{
+    int rc;
+
+    rc = os_mbuf_append(ctxt->om, &ble_svc_audio_pacs_source_audio_locations,
+                        sizeof ble_svc_audio_pacs_source_audio_locations);
+
+    return rc == 0 ? 0 : BLE_ATT_ERR_INSUFFICIENT_RES;
+}
+
+static int
+ble_svc_audio_pacs_find_avail_ctx(uint16_t conn_handle)
```

Comment:
Does this function performs the operation as it's name implies? For me it returns an array index.
Consider returning `struct available_ctx *` from this function